portfolio/src/main/webapp/script.js

```diff
@@ -11,7 +11,65 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
- 
+
+
+// These variables will be used to track whenever a reply button is clicked
+// And whether the same button was clicked again in a row.
+var replyClickCounter = 0;
+var currentReplyBtnClicked = 0;
+
+// Post event listener to submit posts with Fetch on form submit. 
+document.getElementById('post-form').addEventListener('submit', function(postSubmitEvent) {
+  // Prevent form from redirecting to the server.
+  postSubmitEvent.preventDefault();
+  
+  // Get a reference to all the elements on the form.
+  const formData = new FormData(this);
+  const postFormParams = new URLSearchParams();
+  
+  // Append a pair [parameter name: value] of the form's elements
+  // to a URLSearchParams.
+  for (const pair of formData) {
+    postFormParams.append(pair[0], pair[1]);
+  }
```

Comment:
nit: This part is a little confusing because you are appending Form data to URLSearchParams, but then passing it in the body of the POST method which is where Form data would normally go. Typically, URLSearchParams are appended to the end of the URL, and are not passed as part of the body of the message.

I think it would make more sense to append the URLSearchParams to formData because you are ultimately sending the data the way Form data would normally be sent. I don't know if this is actually possible with these objects, but it looks like it should be with the code you have.

Reply:
I tried passing the form data itself but the server was not able to read the values (returning null). And i googled and saw that it's possible to pass the form data to a URLSearchParams, which worked. For time's sake i didn't try to find why form data wasn't working but i think I needed to specify some headers or probably use json.stringify 
I can try that later